cmd/device/update/update.go

```diff
@@ -0,0 +1,162 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+
+	"github.com/edgexfoundry-holding/edgex-cli/config"
+	"github.com/edgexfoundry-holding/edgex-cli/pkg/editor"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/urlclient/local"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/spf13/cobra"
+)
+
+const deviceTemp = `{
+  "Id": "{{.Id}}",
```

Comment:
With update we need to ensure that the user specified Id(s) exist.
Do we allow changing of the name?

Reply:
yes, update of the name is allowed. On the background, the device service is recreated

Reply:
DeviceService name is allowed to be updated - on the background DS is recreated.
That is why I did the same for name. But I did double check and I found that it is impossible to update the name of the device. Probably because the name is used as key in other microservices as well. This totally make sense 
But then isn't it problematic to update the name of DS ?! 

I was about to remove the {ID} from the template but it figured out that I cannot modify the device if do not provide the Id as well , next to the name. (because of this [line](https://github.com/edgexfoundry/edgex-go/blob/e46f0634e742b47d68bad1447e074d74297b9715/internal/core/metadata/operators/device/update.go#L76))This looks suspicious. I will double check this as well 

https://github.com/edgexfoundry/edgex-go/blob/e46f0634e742b47d68bad1447e074d74297b9715/internal/core/metadata/operators/device/update.go#L45-L82